ldes-fragmentisers/ldes-fragmentiser-queue-in-memory/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/mediator/InMemoryFragmentationMediator.java

```diff
@@ -0,0 +1,45 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.fragmentisers.mediator;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesfragment.services.FragmentationExecutor;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesfragment.services.FragmentationMediator;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class InMemoryFragmentationMediator implements FragmentationMediator {
+	private static final Logger LOGGER = LoggerFactory.getLogger(FragmentationMediator.class);
+	private final ExecutorService executorService = Executors.newSingleThreadExecutor();
+	private final LinkedBlockingQueue<String> ldesMembersToFragment = new LinkedBlockingQueue<>();
+
+	private final FragmentationExecutor fragmentationExecutor;
+	protected final AtomicInteger ldesMembersToFragmentTracker;
+
+	public InMemoryFragmentationMediator(FragmentationExecutor fragmentationExecutor,
+			MeterRegistry meterRegistry) {
+		LOGGER.info("Server has been configured to queue ldes members for fragmentation IN MEMORY");
+		this.fragmentationExecutor = fragmentationExecutor;
+		ldesMembersToFragmentTracker = meterRegistry.gauge("ldes_server_members_to_fragment", new AtomicInteger(0));
+	}
+
+	@Override
+	public void addMemberToFragment(String ldesMember) {
+		ldesMembersToFragment.add(ldesMember);
+		processMember(ldesMember);
+	}
+
+	@Override
+	public void processMember(String ldesMember) {
+		executorService.submit(() -> fragmentationExecutor.executeFragmentation(ldesMembersToFragment.poll()));
+	}
+
+	@Scheduled(fixedDelay = 1000)
+	protected void reportWaitingMembers() {
+		ldesMembersToFragmentTracker.set(ldesMembersToFragment.size());
+	}
```

Comment:
Since scheduling is not used elsewhere in the domain, you could also move the SchedulingConfig to this module.

Reply:
This scheduler was originally introduced for logging, this can now be removed entirely since the tracker can be put on the methods when the queue is called